packages/core/src/index.ts

```diff
@@ -173,138 +174,126 @@ export const initialize = <T extends Record<string, any> = Record<string, any>>(
   appsConfig,
   api,
   options,
+  pluginStoreFeatureFlags = {},
+  pluginLoaderOptions = {},
+  pluginStoreOptions = {},
 }: {
   appsConfig: AppsConfig;
   api?: T;
   options?: Partial<ScalprumOptions>;
-}): void => {
+  pluginStoreFeatureFlags?: FeatureFlags;
+  pluginLoaderOptions?: Partial<PluginLoaderOptions>;
+  pluginStoreOptions?: Partial<PluginStoreOptions>;
+}): Scalprum<T> => {
+  if (scalprum) {
+    return scalprum as Scalprum<T>;
+  }
   const defaultOptions: ScalprumOptions = {
     cacheTimeout: 120,
     ...options,
   };
-  window[GLOBAL_NAMESPACE] = {
+
+  // Create new plugin loader instance
+  const pluginLoader = new PluginLoader({
+    ...pluginLoaderOptions,
+    sharedScope: getSharedScope(),
+    getPluginEntryModule: ({ name }) => (window as { [key: string]: any })[name],
+  });
+
+  // Create new plugin store
+  const pluginStore = new PluginStore(pluginStoreOptions);
+  pluginLoader.registerPluginEntryCallback();
```

Comment:
FYI, this is needed only if you plan to support loading plugins using the `callback` registration method.

If all of your plugins are expected to be built exclusively with e.g. `var` library type (and thus using the `custom` registration method), you technically don't need this.

Overall, I'd keep it here unless you're absolutely sure about your plugin registration method expectations.

Reply:
Yes, we plan to support it.